neo/io/nwbio.py

```diff
@@ -620,26 +681,56 @@ def _write_event(self, nwbfile, event):
         if hasattr(event, 'proxy_for') and event.proxy_for == Event:
             event = event.load()
         hierarchy = {'block': segment.block.name, 'segment': segment.name}
-        tS_evt = AnnotationSeries(
-            name=event.name,
-            data=event.labels,
-            timestamps=event.times.rescale('second').magnitude,
-            description=event.description or "",
-            comments=json.dumps(hierarchy))
+        # if constant timestamps
+        timestamps = event.times.rescale('second').magnitude
+        if any(timestamps) == any(timestamps):
+            tS_evt = TimeSeries(name=event.name,
+                                data=event.labels,
+                                starting_time=0.0,
+                                rate=0.01,
+                                unit=str(event.units),
+                                description=event.description or "",
+                                comments=json.dumps(hierarchy))
+        else:
+            tS_evt = TimeSeries(name=event.name,
+                                data=event.labels,
+                                timestamps=event.times.rescale('second').magnitude,
+                                unit=str(event.units),
+                                description=event.description or "",
+                                comments=json.dumps(hierarchy))
+
         nwbfile.add_acquisition(tS_evt)
         return tS_evt
 
-    def _write_epoch(self, nwbfile, epoch):
-        segment = epoch.segment
+    def _write_manage_epoch(self, nwbfile, segment, epoch, arr):
         if hasattr(epoch, 'proxy_for') and epoch.proxy_for == Epoch:
             epoch = epoch.load()
         for t_start, duration, label in zip(epoch.rescale('s').magnitude,
                                             epoch.durations.rescale('s').magnitude,
-                                            epoch.labels):
-            nwbfile.add_epoch(t_start, t_start + duration, [label], [],
-                              _name=epoch.name,
-                              segment=segment.name,
-                              block=segment.block.name)
+                                            epoch.labels,
+                                            ):
+            for j in [label]:
+                t_stop = t_start + duration
+                seg_name = "%s %s" % (epoch.segment.name, label)
```

Comment:
Here occurs the same error as with the events earlier. You have to keep a reference to the segment from before the proxy object is loaded.